spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/KafkaStreamsMessageConversionDelegate.java

```diff
@@ -105,25 +105,24 @@ public KStream deserializeOnInbound(Class<?> valueClass, KStream<?, ?> bindingTa
 				boolean isValidRecord = false;
 
 				try {
-					if (valueClass.isAssignableFrom(o2.getClass())) {
-						keyValueThreadLocal.set(new KeyValue<>(o, o2));
-					}
-					else if (o2 instanceof Message) {
-						if (valueClass.isAssignableFrom(((Message) o2).getPayload().getClass())) {
-							keyValueThreadLocal.set(new KeyValue<>(o, ((Message) o2).getPayload()));
-						}
-						else {
-							convertAndSetMessage(o, valueClass, messageConverter, (Message) o2);
+					//if the record is a tombstone, ignore and exit from processing further.
+					if (o2 != null) {
```

Comment:
Don't we need to do anything in the else?
Is logic in the end of the method going to work if we don't add anything into the `ThreadLocal` holder?

The `Object objectValue = keyValueThreadLocal.get().value;` is going to fail with `NPE`. No?

Reply:
The last block in the method will only be invoked if there is a stream coming through that branch. The first filter above that returns `false`, thus there is no stream coming through `branch[0]` and therefore it won't be invoked. So, I think we are safe there.

Reply:
OK. So, I need to understand first of all what is that `branch()` for. My home work though 😄 

Meanwhile maybe it make sense at least log INFO why have we stopped processing here. 
Does it make sense?

Reply:
`else` case is added with logging.